### ISSUE TITLE
Test failure on 32-bit Linux: ALTER SUBSCRIPTION$1 SET PUBLICATION insert_only

### ISSUE THREAD

**Issue by df7cb** [pganalyze/libpg_query#158]
libpg_query 13-2.2.0 is failing the testsuite on 32-bit Debian/Ubuntu:

```
16:53:37 ..............INVALID result for "ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only"
16:53:37 expected: ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only
16:53:37 actual: ALTER SUBSCRIPTION$1 SET PUBLICATION insert_only
16:53:37 INVALID result for "ALTER SUBSCRIPTION mysub CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb'"
16:53:37 expected: ALTER SUBSCRIPTION mysub CONNECTION $1
16:53:37 actual: ALTER SUBSCRIPTION$1 CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb'
16:53:37 .....
16:53:37 Makefile:245: recipe for target 'test' failed
```

Full build log: https://pgdgbuild.dus.dg-i.net/job/libpg-query-binaries/architecture=i386,distribution=bionic/10/console
https://pgdgbuild.dus.dg-i.net/job/libpg-query-binaries/

**Comment by lfittl** [comment]
@df7cb Thanks for the report - thats quite odd, since the test suite runs just fine here (although that only tests 64-bit today). Are there any modifications to the libpg_query source?

**Comment by df7cb** [comment]
There are, but none that seem related: https://salsa.debian.org/postgresql/libpg-query/-/tree/master/debian/patches

**Comment by lfittl** [comment]
Thanks - might be an issue with the recent normalization changes. I'll try to get a 32-bit
environment running later today to reproduce.

**Comment by jirutka** [comment]
I ran into the same problem on Alpine Linux Edge x86, armhf, and armv7. We don’t modify the libpg_query source, just [unbundle some dependencies](https://gitlab.alpinelinux.org/alpine/aports/-/blob/4b3afccdef4a731fb49b57b609df3190edf408c9/community/libpg_query/unbundle-vendor-deps.patch).

**Comment by df7cb** [comment]
Fwiw the problem is still present in 14-3.0.0:

```
16:23:39 test/normalize
16:23:39 ..............INVALID result for "ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only"
16:23:39 expected: ALTER SUBSCRIPTION mysub SET PUBLICATION insert_only
16:23:39 actual: ALTER SUBSCRIPTION$1 SET PUBLICATION insert_only
16:23:39 INVALID result for "ALTER SUBSCRIPTION mysub CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb'"
16:23:39 expected: ALTER SUBSCRIPTION mysub CONNECTION $1
16:23:39 actual: ALTER SUBSCRIPTION$1 CONNECTION 'host=192.168.1.50 port=5432 user=foo dbname=foodb'
16:23:39 .....
```

**Comment by lfittl** [comment]
@df7cb Thanks for checking - its still on my list to get to the bottom of the issue here, as the relevant code portions are in the libpg_query code itself, not the original Postgres code.

Hoping to get some time for this latest in the down period around the holidays. For context, we don't have a use case for 32-bit support in pganalyze, so this is on the back burner until I have more time for OSS maintenance work -- patches of course welcome to address sooner.

**Comment by lfittl** [comment]
32-bit support should be resolved with the 16-5.1.0 release (https://github.com/pganalyze/libpg_query/releases/tag/16-5.1.0), thus closing this.
